engine/src/main/scala/cromwell/webservice/PartialWorkflowSources.scala

```diff
@@ -20,51 +20,54 @@ final case class PartialWorkflowSources(workflowSource: Option[WorkflowSource],
-      } else if (name == "workflowInputs") {
-        partialSources.copy(workflowInputs = workflowInputs(data.utf8String))
-      } else if (name.startsWith("workflowInputs_")) {
-        val index = name.stripPrefix("workflowInputs_").toInt
-        partialSources.copy(workflowInputsAux = partialSources.workflowInputsAux + (index -> data.utf8String))
-      } else if (name == "workflowOptions") {
-        partialSources.copy(workflowOptions = Option(data.utf8String))
-      } else if (name == "wdlDependencies" || name == "workflowDependencies") {
-        if (name == "wdlDependencies") deprecationWarning(out = "wdlDependencies", in = "workflowDependencies")
-        partialSources.copy(zippedImports = Option(data.toArray))
-      } else if (name == "customLabels") {
-        partialSources.copy(customLabels = Option(data.utf8String))
-      } else {
-        throw new IllegalArgumentException(s"Unexpected body part name: $name")
+  def fromSubmitRoute(formData: Map[String, ByteString],
+                      allowNoInputs: Boolean): Try[Seq[WorkflowSourceFilesCollection]] = {
+    val partialSources: Try[PartialWorkflowSources] = Try {
+      formData.foldLeft(PartialWorkflowSources.empty) {
+        (partialSources: PartialWorkflowSources, kv: (String, ByteString)) =>
+          val (name, data) = kv
+
+          name match {
+            case "wdlSource" =>
+              val warning = deprecationWarning(out = "wdlSource", in = "workflowSource")
+              val warnings = warning +: partialSources.warnings
+              partialSources.copy(workflowSource = Option(data.utf8String), warnings = warnings)
+            case "workflowSource" => partialSources.copy(workflowSource = Option(data.utf8String))
+            case "workflowType" => partialSources.copy(workflowType = Option(data.utf8String))
+            case "workflowTypeVersion" => partialSources.copy(workflowTypeVersion = Option(data.utf8String))
+            case "workflowInputs" => partialSources.copy(workflowInputs = workflowInputs(data.utf8String))
+            case _ if name.startsWith("workflowInputs_") =>
+              val index = name.stripPrefix("workflowInputs_").toInt
+              partialSources.copy(workflowInputsAux = partialSources.workflowInputsAux + (index -> data.utf8String))
+            case "workflowOptions" => partialSources.copy(workflowOptions = Option(data.utf8String))
+            case "wdlDependencies" =>
+              val warning = deprecationWarning(out = "wdlDependencies", in = "workflowDependencies")
+              val warnings = warning +: partialSources.warnings
+              partialSources.copy(zippedImports = Option(data.toArray), warnings = warnings)
+            case "workflowDependencies" => partialSources.copy(zippedImports = Option(data.toArray))
+            case "customLabels" => partialSources.copy(customLabels = Option(data.utf8String))
+            case _ => throw new IllegalArgumentException(s"Unexpected body part name: $name")
```

Comment:
ToL, NYF: I think this might be a better UX as a warning?

Reply:
Someday. This line was only touched after a request to switch the existing `if` statements to `case` matches.